### ISSUE TITLE
Unneeded header generation

### ISSUE THREAD

**Issue by jcfreeman2** [DUNE-DAQ/daq-cmake#49]
Kurt originally brought this issue to my attention today using the full stack of packages under development, and I've been able to recreate it using only appfwk and opmonlib in my work area. Essentially, what's happening is that after a full, successful build, if you run `dbt-build.sh` again without touching any code - which _logically_ should be a no-op - it will alternate between (A) just copying directories into the build area but not generating any code or building anything else, and (B) generating headers using opmonlib's `InfoStructs.hpp.j2` template and thus triggering pretty much a full rebuild.

Poking around, the thing that distinguishes `InfoStructs.hpp.j2` from `InfoNljs.hpp.j2` is that it includes `info_hpp_macros.j2`, and indeed if I also have `InfoNljs.hpp.j2` include it, then it too will be used to generate headers in this on again / off again manner. So that's a clue to work off of.

**Comment by jcfreeman2** [comment]
The discussion on Issue #39 should also help shed some light on what's going on

**Comment by jcfreeman2** [comment]
What seems to be happening is this. When a build occurs, a file is created by the `moo render-deps` command and placed in the build area with a name like, e.g., `InfoStructs.hpp.d`. This type of file lists files which, if modified, trigger a fresh header generation. In the case of `InfoStructs.hpp.d`, it lists `info_hpp_macros.j2`, but if you've got opmonlib as one of the packages you're building (as opposed to a ups product) then `InfoStructs.hpp.d` will list the `info_hpp_macros.j2` in your build area _which gets freshly copied into the build area every time you build_.  I've added a PR where instead of looking at template files in the build area, it checks to see if the template files in the source code area have been modified. 

An aside: there's another issue I encountered where it seems like not all the files you'd expect to be in these dependency files are listed, but that's a separate phenomenon to what was going on here. I'll look into it.
